### DISCUSSION TITLE
Meeting Lists Brings Up Wrong Day.

### DISCUSSION THREAD

**Discussion by Grateful-Viv** [code4recovery/12-step-meeting-list#800] [Q&A] [answered]
This issue has been going on and off, but more on beginning in the past couple of weeks.

People go to our website to find a meeting. There’s a big fat button on the page that says find a meeting. This is the link: https://sussexaa.org/meetings/

But it’s not bringing up the correct day. Without realizing it they’re scrolling down the list and don’t see the meeting that they’re looking for. Then they go to the top and select the right day (which people didn’t need to do before) and it doesn’t show up the correct day until they hit refresh. And sometimes refresh makes no difference. It’s stuck on the wrong day. 

At first I thought it was a user error or two or three, but it is happening with increasing regularity. 

This is happening on iOS devices using safari. I thought it was their browser cache that needed refreshing, but it’s happening to too many different people. 

The site is using the legacy UI and version 3.14.2. I can verify the same issue they’re seeing. 

It’s almost like it’s not syncing with today’s date and time because even when I go to this link: https://sussexaa.org/meetings/?tsml-time=upcoming it shows meetings for starting at 630 PM. But right now it’s 9:25pm and we have no more meetings today. 

Refreshing makes no difference. 

The Upcoming Meetings Widget is also out of sync. It’s showing meetings from 8 this morning instead of displaying no more meetings today. 

When I view same meeting data on my staging site using TSML Ui, all shows correct. I’d like to start using TSML and have just been waiting for some changes. 

i’ve been hearing a lot of complaints and it’s really frustrating to figure out what’s going on. I’m gonna tell people to start using the Meeting Guide app and not the website until we can get this resolved.

**Comment by joshreisner** [comment, marked as answer]
hi viv, yes i see it's showing Tuesday for me on your website. i'm not aware of this happening other places. it's working properly on our demo site for example.

do you have a caching plugin installed? i suspect that's probably what's going on, since the time it's showing is in the past.

sometimes websites have the timezone set incorrectly (on wordpress general settings) but in those cases the time is usually ahead.

**Reply by Grateful-Viv** [reply]
I am using SiteGround Optimizer. The settings are on for Dynamic Caching, File-Based Caching, and Memcaching. I believe I recently turned on File-Based Caching. I am going to turn that off. But can you explain why TSML UI is great with showing correct date/times and Legacy is not?

The weird thing is this doesn't happen on my desktop.

**Reply by joshreisner** [reply]
i'm not sure why it wouldn't happen on your desktop… are you logged in to WP when it displays properly? perhaps optimizer disables caching for logged-in users.

the reason TSML UI works is it's getting today's date from your browser, rather than from the server. the caching mediates between TSML and the server, but it doesn't mediate between the frontend and your browser, if that makes sense.

**Reply by Grateful-Viv** [reply]
Would turning off file based caching make a difference? 

And yes, I am logged into the WP Admin when it's showing properly. I suppose I could also check on my laptop without being logged in.

**Reply by joshreisner** [reply]
it may also be that when you make a change to the meeting list, it refreshes the cache, which fixes the page temporarily

yes i'm sure the caching is the culprit. turning it off should let things get back to normal. 

if you don't want to turn it all the way off, perhaps there is a way to exclude the meetings page from caching?

**Reply by Grateful-Viv** [reply]
I can exclude URLs from caching by simply typing in that section /meetings/*

**Reply by joshreisner** [reply]
i hope that fixes it 🤞

**Reply by Grateful-Viv** [reply]
This morning it displayed Wednesday meetings. I have a few people now checking this on their devices. So in addition to turning off file cache, I did add /meetings/* to the exclude URLs from caching section. Hopefully 🤞🏼this will do it. Thanks Josh for stepping in so quickly.

**Comment by Grateful-Viv** [comment]
Wanted to report that it’s far so good after turning off file caching and excluding all meeting URLs.
